service/core/kafka/client/src/main/scala/com/lightbend/lagom/internal/broker/kafka/KafkaSubscriberActor.scala

```diff
@@ -46,41 +45,20 @@ private[lagom] class KafkaSubscriberActor[Payload, SubscriberPayload](
     extends Actor
     with ActorLogging {
 
-  /** Switch used to terminate the on-going Kafka publishing stream when this actor fails.*/
-  private var shutdown: Option[KillSwitch] = None
-
   override def preStart(): Unit = {
-    kafkaConfig.serviceName match {
-      case Some(name) =>
-        log.debug("Looking up Kafka service from service locator with name [{}] for at least once source", name)
-        locateService(name)
-          .map {
-            case Nil  => None
-            case uris => Some(UriUtils.hostAndPorts(uris))
-          }
-          .pipeTo(self)
-        context.become(locatingService(name))
-      case None =>
-        run(None)
-    }
-  }
-
-  override def postStop(): Unit = {
-    shutdown.foreach(_.shutdown())
-  }
-
-  private def locatingService(name: String): Receive = {
-    case Status.Failure(e) =>
-      log.error(e, s"Error locating Kafka service named [{}]", name)
-      throw e
+    val drainingControl: DrainingControl[Done] =
+      atLeastOnce()
+        .toMat(Committer.sink(consumerConfig.committerSettings))(Keep.both)
+        .mapMaterializedValue(DrainingControl.apply[Done])
```

Comment:
With Alpakka Kafka 2.0.3 you can use the `DrainingControl.apply[Done]` instead of `Keep.both`.

Reply:
Thanks! Done ✅